icubam/predicu/data.py

```diff
@@ -233,14 +232,99 @@ def aggregate_multiple_inputs(d):
   return pd.concat(res_dfs)
 
 
+def fill_in_missing_days(d, time_delta_threshold="3D"):
+  res_dfs = []
+  for icu_name, dg in d.groupby("icu_name"):
+    dg = dg.sort_values(by=["datetime"])
+    time_delta = dg["datetime"].diff(1)
+    for i, td in enumerate(time_delta):
+      if td > pd.Timedelta(time_delta_threshold):
+        n_days = td // pd.Timedelta("1D")
+        val_init = dg.iloc[i - 1]
+        val_final = dg.iloc[i]
+        for added_day in range(n_days):
+          added_datetime = (val_init.datetime + pd.Timedelta("1D") * added_day)
+          added_date = val_init.date + pd.Timedelta("1D") * added_day
+          new_row = {
+            "datetime":
+            added_datetime,
+            "icu_name":
+            val_init.icu_name,
+            "date":
+            added_date,
+            "department":
+            val_init.department,
+            "n_covid_deaths":
+            np.round(
+              val_init.n_covid_deaths +
+              (val_final.n_covid_deaths - val_init.n_covid_deaths) *
+              added_day * 1.0 / n_days,
+              4,
+            ),
+            "n_covid_healed":
+            np.round(
+              val_init.n_covid_healed +
+              (val_final.n_covid_healed - val_init.n_covid_healed) *
+              added_day * 1.0 / n_days,
+              4,
+            ),
+            "n_covid_transfered":
+            np.round(
+              val_init.n_covid_transfered +
+              (val_final.n_covid_transfered - val_init.n_covid_transfered) *
+              added_day * 1.0 / n_days,
+              4,
+            ),
+            "n_covid_refused":
+            np.round(
+              val_init.n_covid_refused +
+              (val_final.n_covid_refused - val_init.n_covid_refused) *
+              added_day * 1.0 / n_days,
+              4,
+            ),
+            "n_covid_free":
+            np.round(
+              val_init.n_covid_free +
+              (val_final.n_covid_free - val_init.n_covid_free) * added_day *
+              1.0 / n_days,
```

Comment:
I think we should be able to do this with `Series.interpolate` (or at least factorize repeated code into a function).

Reply:
yes definitely